### PR TITLE
Decode a Morse code message

### PR DESCRIPTION
**Pull request by Hisoka37**
### In this PR Me and My partner We : 

- [ ] Create a method to decode a Morse code character, takes a string parameter, and return the corresponding character in uppercase (e.g. decode_char(".-") returns "A").
- [ ] Create a method to decode an entire word in Morse code, takes a string parameter, and return the string representation. Every character in a word will be separated by a single space (e.g. decode_word("-- -.--") returns "MY").
- [ ] Create a method to decode the entire message in Morse code, takes a string parameter, and return the string representation. Every word will be separated by 3 spaces (e.g.
decode("-- -.-- -. .- -- .").
- [ ] Reuse methods whenever possible. Make sure that your code is DRY.
- [ ] Once your code is ready use it to decode the message from the old bottle:
.- -... --- -..- ..-. ..- .-.. .-.. --- ..-. .-. ..- -... .. . ...

Cheers and Happy Coding 😸

### CODE REVIEW

**Review by DammyShittu — COMMENTED** [review]
Hi @Hisoka37 and @Odongo006,

While you made a great effort in this project, I cannot proceed to review your code.

## Invalid Code Review Request

### You are required to request a merge to develop branch, not main according to the gitflow

Your Code Review Request will be marked as invalid in your Dashboard, so please submit a new one once you are ready 🙏

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

--------

_Invalid Code Review Request **does not count** into the code reviews limit._

File: morse.rb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  MORSE_CODE[char]
+end
+
+def decode(line)
+  words = line.split('   ')
+  decoded_words = words.map do |word|
+    word.split.map { |char| decode_char(char) }.join
+  end
+  decoded_words.join(' ')
+end
```
Comment (amejid) [inline morse.rb]: - [ ] You did a great job creating the method to decode Morse code. :+1: There seems to be one function missing, which is supposed to decode a word. Please implement such a function. You can move some of the lines from the `decode` function to achieve it. 😎️

**Requirements**
image

**Review by amejid — CHANGES_REQUESTED** [review]
Hi team, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



## Highlights
- Linter check passing :heavy_check_mark: 
- Correct workflow :heavy_check_mark: 
- Function to decode a character and a message implemented :heavy_check_mark: 

## Required Changes ♻️
- There seems to be a merge conflict between branches. Please attend to it.

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: morse.rb (hunk 1 above)
Comment (Hisoka37) [reply morse.rb]: Hi @amejid In our code if you want to decode word("-- -.--") it will return "MY" , i don't understand what you are asking for?

Comment (amejid) [reply morse.rb]: Hey @Hisoka37, apparently the `decode` function in your project does both things, it decodes a word and sentence. Please refactor and break it up into two functions, one to decode a word and one to decode an entire message. The requirement suggests that we should have three separate functions. Happy coding. 🤝️

**Project requirements**
image

**Review by IngDavidTM — APPROVED** [review]
Hi Team 🙋‍♂️, 

# ✔️   Status: APPROVED 🏆 🎉 🎊 

You have done a great job on this project, keep it up 👍 🥇. Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations!  🎉💯🌟 

## Top highlight 🎯

- The Pull Request follows the GitFlow ✅ 
- The PR title and description are professional and descriptive ✅ 
- Linters are working properly ✅ 
- Function to decode a character and a message implemented ✅ 


## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**